internal/controller/issues.go

```diff
@@ -0,0 +1,180 @@
+	}
+
+	PRChange := r.CheckForPr(githubIssue, issue)
+	OpenChange := r.CheckIfOpen(githubIssue, issue)
+
+	if PRChange || OpenChange {
+		r.Log.Info("Updating Issue status", zap.String("IssueName", issue.Name), zap.String("Namespace", issue.Namespace))
+
+		if err := r.Client.Status().Update(ctx, issue); err != nil {
+			r.Log.Warn("Status update failed, attempting fallback", zap.Error(err))
+
+			if fallbackErr := r.Client.Update(ctx, issue); fallbackErr != nil {
+				r.Recorder.Event(issue, corev1.EventTypeWarning, "StatusUpdateFailed", fmt.Sprintf("Failed to update status: %v", fallbackErr))
+				return fmt.Errorf("failed to update status: %v", fallbackErr)
+			}
+		}
+
+		r.Recorder.Event(issue, corev1.EventTypeNormal, "StatusUpdated", "Issue status updated successfully")
+		r.Log.Info("Issue status updated successfully", zap.String("IssueName", issue.Name), zap.String("Namespace", issue.Namespace))
+	} else {
+		r.Log.Info("No changes detected in issue status", zap.String("IssueName", issue.Name), zap.String("Namespace", issue.Namespace))
+	}
+
+	return nil
+}
+
+// CheckIfOpen checks if the issue is open
+func (r *GithubIssueReconciler) CheckIfOpen(githubIssue *github.Issue, issueObject *issues.GithubIssue) bool {
+	if githubIssue == nil {
+		return false
+	}
+	state := githubIssue.GetState()
+	condition := &metav1.Condition{
+		Type:    "IssueIsOpen",
+		Status:  metav1.ConditionTrue,
+		Reason:  "IssueIsOpen",
+		Message: "Issue is open",
+	}
+	if state != "open" {
+		condition = &metav1.Condition{
+			Type:    "IssueIsOpen",
+			Status:  metav1.ConditionFalse,
+			Reason:  fmt.Sprintf("IssueIs%s", state),
+			Message: fmt.Sprintf("Issue is %s", state),
+		}
+	}
+	if !meta.IsStatusConditionPresentAndEqual(issueObject.Status.Conditions, "IssueIsOpen", condition.Status) {
+		meta.SetStatusCondition(&issueObject.Status.Conditions, *condition)
+		return true
+	}
+	return false
+}
+
+// CheckForPr checks if the issue has an open PR
+func (r *GithubIssueReconciler) CheckForPr(githubIssue *github.Issue, issueObject *issues.GithubIssue) bool {
+	if githubIssue == nil {
+		return false
+	}
+	condition := &metav1.Condition{
+		Type:    "IssueHasPR",
+		Status:  metav1.ConditionFalse,
+		Reason:  "IssueHasNoPR",
+		Message: "Issue has no PR",
+	}
+	if githubIssue.GetPullRequestLinks() != nil {
+		condition = &metav1.Condition{
+			Type:    "IssueHasPR",
+			Status:  metav1.ConditionTrue,
+			Reason:  "IssueHasPR",
+			Message: "Issue has an open PR",
+		}
+	}
+	if !meta.IsStatusConditionPresentAndEqual(issueObject.Status.Conditions, "IssueHasPR", condition.Status) {
+		meta.SetStatusCondition(&issueObject.Status.Conditions, *condition)
+		return true
+	}
+	return false
+}
+
+// fetchAllIssues gets all issues in a repository with retry for rate limits
+func (r *GithubIssueReconciler) fetchAllIssues(ctx context.Context, owner string, repo string) ([]*github.Issue, error) {
+	opt := &github.IssueListByRepoOptions{}
```

Comment:
This is empty with all struct fields having zero values so why are you even setting it?